ggshield/iac/iac_scan_models.py

```diff
@@ -37,34 +45,95 @@ class IaCDiffScanResult(Base):
-    # TODO: move this into GGClient
-    def mock_api_iac_diff_scan(
+    def post(
+        self,
+        endpoint: str,
+        json: Optional[Dict[str, Any]] = None,
+        extra_headers: Optional[Dict[str, str]] = None,
+        **kwargs: Any,
+    ) -> Response:
+        return self.request(
+            "post",
+            endpoint=endpoint,
+            json=json,
+            extra_headers=extra_headers,
+            **kwargs,
+        )
+
+    def request(
+        self,
+        method: str,
+        endpoint: str,
+        version: Optional[str] = DEFAULT_API_VERSION,
+        extra_headers: Optional[Dict[str, str]] = None,
+        **kwargs: Any,
+    ) -> Response:
+        url = self._url_from_endpoint(endpoint, version)
+
+        headers = (
+            {**self.session.headers, **extra_headers}
+            if extra_headers
+            else self.session.headers
+        )
+        start = time.time()
+        response: Response = self.session.request(
+            method=method, url=url, timeout=self.timeout, headers=headers, **kwargs
+        )
+        duration = time.time() - start
+        logger.debug(
+            "method=%s endpoint=%s status_code=%s duration=%f",
+            method,
+            endpoint,
+            response.status_code,
+            duration,
+        )
+
+        self.app_version: Optional[str] = response.headers.get(
+            "X-App-Version", self.app_version
+        )
+        self.secrets_engine_version: Optional[str] = response.headers.get(
+            "X-Secrets-Engine-Version", self.secrets_engine_version
+        )
+        return response
+
+    # TODO: Move this method into GGClient.
+    def api_iac_diff_scan(
         self,
         reference: bytes,
         current: bytes,
         scan_parameters: IaCScanParameters,
         extra_headers: Optional[Dict[str, str]] = None,
     ) -> Union[Detail, IaCDiffScanResult]:
-        scan = self.iac_directory_scan(
-            Path("."),
-            [],
-            scan_parameters,
-            extra_headers,
-        )
-        result = IaCDiffScanResult(
-            scan.id,
-            scan.type,
-            scan.iac_engine_version,
-            entities_with_incidents=IaCDiffScanEntities([], [], []),
-        )
-        result.status_code = 200
+        result: Union[Detail, IaCDiffScanResult]
+        try:
+            resp = self.post(
+                endpoint="iac_diff_scan",
+                extra_headers=extra_headers,
+                files={
+                    "reference": reference,
+                    "current": current,
+                },
+                data={
+                    "scan_parameters": IaCScanParametersSchema().dumps(scan_parameters),
+                },
+            )
+        except requests.exceptions.ReadTimeout:
+            result = Detail("The request timed out.")
+            result.status_code = 504
+        else:
+            result = IaCDiffScanResultSchema().load(resp.json())
```

Comment:
Assuming `IaCDiffScanResult` inherits from `FromDictMixin`, you can use `result = IacDiffScanResult.from_dict(resp.json())` here.